### PR TITLE
Modernise units related frontend code

### PR DESCRIPTION
**Pull request by akheron**
#### Summary

Refactor UnitContext. Remove the scroll position tracking because it didn't work. A more general solution is needed in the future, I think.

Use renderResult, useApiState, add some React.memo calls, etc.

### CODE REVIEW

File: frontend/src/employee-frontend/components/unit/TabGroups.tsx [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+import { UnitContext } from '../../state/unit'
+import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
+import Groups from '../../components/unit/tab-groups/Groups'
+import MissingGroupPlacements from '../../components/unit/tab-groups/MissingGroupPlacements'
+import { combine, Result } from 'lib-common/api'
+import { Action } from 'lib-common/generated/action'
+
 {
   /*
 SPDX-FileCopyrightText: 2017-2020 City of Espoo
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/unit/TabGroups.tsx:13]: Move the licence stuff to the top.

File: frontend/src/employee-frontend/components/unit/TabWaitingConfirmation.tsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { renderResult } from '../async-rendering'
+
 {
   /*
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/unit/TabWaitingConfirmation.tsx:4]: Same here.

**Review by jaahon — COMMENTED** [review]
Good stuff 🏅 just clean up some of the licencing comments.

File: frontend/src/employee-frontend/components/unit/TabGroups.tsx (hunk 1 above)
Comment (akheron) [reply frontend/src/employee-frontend/components/unit/TabGroups.tsx]: Good catch! Fixed license comment style in all .tsx files now.

**Review by jaahon — APPROVED** [review]

**Pull request merged**